provider/cmd/pulumi-resource-pulumiservice/schema.json

```diff
@@ -1219,6 +1219,47 @@
         "stack",
         "timestamp"
       ]
+    },
+    "pulumiservice:index:Environment": {
+      "description": "An ESC Environment.",
+      "properties": {
+        "organization": {
+          "description": "Organization name.",
+          "type": "string"
+        },
+        "name": {
+          "description": "Environment name.",
+          "type": "string"
+        },
+        "yaml": {
+          "description": "Environment's yaml file.",
+          "$ref": "pulumi.json#/Asset"
```

Comment:
Should make absolutely certain that assets work with output properties. Not 100% sure about that.

Reply:
Confirmed it works with output using Random and below snippet of code:

```
var random = new RandomString("rand", new RandomStringArgs
    {
        Length = 5,
        Special = false,
    });

    Output<AssetOrArchive> asset = random.Result.Apply(res => {
        String yaml = """
            imports:
              - dev-stacks
            values:
              aws:
                secrets:
                  fn::open::aws-secrets:
                    region: us-west-2
                    login: ${aws.creds}
                    get:
                      allSecrets:
                        secretId: iaro-dev-stack/pulumi-service
              secrets:
                fn::fromJSON: ${aws.secrets.allSecrets}
              pulumiConfig:
                gitHubOAuthID: ${secrets.gitHubOAuthID}
                gitHubOAuthSecret: ${secrets.gitHubOAuthSecret}
                randomString: 
        """ + res;
        AssetOrArchive asset = new StringAsset(yaml);
        return asset;
        }
    );

    var environ = new Pulumi.PulumiService.Environment(
        "Iaro's environment",
        new EnvironmentArgs {
            Organization = "IaroslavTitov",
            Name = "IaroEnv",
            Yaml = asset
        }
    );
```

When I go to see the environment in console it has a random blurb `zbhrh` and the same in Random's resource properties.

Reply:
Ah, I should clarify--I meant that I'm not sure that the output property `Environment.Yaml` will work with assets. I'd try passing the `Yaml` output of one environment to the `Yaml` input of another.

Reply:
Oooh, gotcha, didn't understand properly
Confirmed that this works:
```
var environ = new Pulumi.PulumiService.Environment(
        "Iaro's environment",
        new EnvironmentArgs {
            Organization = "IaroslavTitov",
            Name = "IaroEnv",
            Yaml = asset
        }
    );

    var environ2 = new Pulumi.PulumiService.Environment(
        "Iaro's environment 2",
        new EnvironmentArgs {
            Organization = "IaroslavTitov",
            Name = "IaroEnv2",
            Yaml = environ.Yaml.Apply(yaml => yaml)
        }
    );
```

Reply:
FWIW @IaroslavTitov you shouldn't have to do this:
```
Yaml = environ.Yaml.Apply(yaml => yaml)
```

It can just be:
```
Yaml = environ.Yaml
```

Reply:
Oh, makes sense, got stuck in the Apply mindset